api/src/main/java/org/apache/cloudstack/api/command/user/vm/DeployVMCmd.java

```diff
@@ -840,6 +863,10 @@ public void execute() {
 
     @Override
     public void create() throws ResourceAllocationException {
+        if (Stream.of(templateId, snapshotId, volumeId).filter(Objects::nonNull).count() != 1) {
+            throw new CloudRuntimeException(String.format("Only one of the parameters - template ID [%s], volume ID [%s] or snapshot ID [%s] - should be provided to deploy a Virtual machine", templateId, volumeId, snapshotId));
```

Comment:
Very minor: instead of logging the values can we log just the parameter names? Since now templateid won't be required this check also covers the case when non of them are set, and would display all null values. Maybe also could be rephased to something like: Please provide only one of the following parameters.. ?